### PR TITLE
Adapt Fast-HotStuff Reference Node Implementation For Integration Testing

### PR DESCRIPTION
**Pull request by tholonious**

### CODE REVIEW

**Comment by tholonious** [comment]
Current dependencies on/for this PR:
* feature/proof-of-stake
  * **PR #649** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/649" target="_blank">Graphite</a> 
    * **PR #720** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/720" target="_blank">Graphite</a> 
      * **PR #721** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/721" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/721?utm_source=stack-comment).

File: consensus/integration_test_types.go [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+	"github.com/holiman/uint256"
+)
+
+// validatorNode is a simplified implementation of a Fast-HotStuff node that runs the Fast-HotStuff
+// event loop. It is adapted into a working state from the reference implementation at:
+// https://github.com/deso-protocol/hotstuff_pseudocode/blob/main/fast_hotstuff_bls.go
+//
+// This node implementation implementation has no networking, and no node syncing. It strictly runs
+// the Fast-HotStuff protocol as a single node. It is used to test the correctness of the Fast-HotStuff
+// event loop in for integration testing.
```
Comment (lazynina) [inline consensus/integration_test_types.go:17]:
```suggestion
// event loop in integration testing.
```

File: consensus/integration_test_types.go [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+}
+
+func newValidatorNode(stake *uint256.Int, isBlockProposer bool) *validatorNode {
+	return &validatorNode{
+		privateKey:      createDummyBLSPrivateKey(),
+		stake:           stake,
+		eventLoop:       NewFastHotStuffEventLoop(),
+		isBlockProposer: isBlockProposer,
+
+		quit: make(chan struct{}),
```
Comment (lazynina) [inline consensus/integration_test_types.go:40]: Should we initialize the `safeBlocks` map?

Comment (lazynina) [reply consensus/integration_test_types.go:40]: Never mind, I see that the `Init` function handles this for us.

**Review by lazynina — APPROVED** [review]

**Pull request merged**